it/controllers/PostcodeLookupControllerISpec.scala

```diff
@@ -0,0 +1,194 @@
+       |        "country": {
+       |          "code": "UK",
+       |          "name": "United Kingdom"
+       |        },
+       |        "subdivision":{
+       |          "code": "GB-SCT",
+       |          "name": "Scotland"
+       |        },
+       |        "status": 0
+       |      },
+       |      "language": "en"
+       |    }
+       |  ]
+       |""".stripMargin
+
+  val designatoryDetails: String =
+    """|
+     |{
+       |  "etag" : "115",
+       |  "person" : {
+       |    "firstName" : "HIPPY",
+       |    "middleName" : "T",
+       |    "lastName" : "NEWYEAR",
+       |    "title" : "Mr",
+       |    "honours": "BSC",
+       |    "sex" : "M",
+       |    "dateOfBirth" : "1952-04-01",
+       |    "nino" : "TW189213B",
+       |    "deceased" : false
+       |  },
+       |  "address" : {
+       |    "line1" : "26 FARADAY DRIVE",
+       |    "line2" : "PO BOX 45",
+       |    "line3" : "LONDON",
+       |    "postcode" : "CT1 1RQ",
+       |    "startDate": "2009-08-29",
+       |    "country" : "GREAT BRITAIN",
+       |    "type" : "Residential",
+       |    "status": 1
+       |  }
+       |}
+       |""".stripMargin
+
+  override def beforeEach(): Unit = {
+    server.resetAll()
+    beforeEachHomeController()
+
+    server.stubFor(
+      get(urlEqualTo(s"/citizen-details/$generatedNino/designatory-details"))
+        .willReturn(ok(designatoryDetails))
+    )
+
+    server.stubFor(
+      get(urlEqualTo(s"/agent-client-authorisation/status"))
+        .willReturn(
+          ok(
+            Json
+              .toJson(
+                AgentClientStatus(
+                  hasPendingInvitations = true,
+                  hasInvitationsHistory = true,
+                  hasExistingRelationships = true
+                )
+              )
+              .toString
+          )
+        )
+    )
+
+    server.stubFor(
+      put(urlMatching(s"/keystore/pertax-frontend/.*"))
+        .willReturn(ok(Json.toJson(CacheMap("id", Map.empty)).toString))
+    )
+
+    server.stubFor(
+      get(urlEqualTo(personDetailsUrl))
+        .willReturn(ok(FileHelper.loadFile("./it/resources/person-details.json")))
+    )
+
+    lazy val featureFlagService = app.injector.instanceOf[FeatureFlagService]
+    featureFlagService.set(TaxcalcToggle, enabled = false).futureValue
+    featureFlagService.set(SingleAccountCheckToggle, enabled = true).futureValue
+    featureFlagService.set(TaxComponentsToggle, enabled = true).futureValue
+    featureFlagService.set(RlsInterruptToggle, enabled = false).futureValue
+  }
+
+  "personal-account" must {
+    "show edit address when the address service times out after 5 secs" in {
+
+      val wholeStreetRequestBody: JsObject = Json.obj(
+        "postcode" -> "AA11AA"
+      )
+
+      server.stubFor(
+        post(urlEqualTo(urlPost))
+          .withRequestBody(equalToJson(wholeStreetRequestBody.toString))
+          .willReturn(ok(addressRecordSet).withFixedDelay(7000))
```

Comment:
You need to overwrite the delay in your test Application above so that your test does not take 5 seconds to run.